src/Service/S3/tests/Unit/Input/PutBucketNotificationConfigurationRequestTest.php

```diff
@@ -0,0 +1,112 @@
+                        'Id' => 'change me',
+                        'TopicArn' => 'change me',
+                        'Events' => ['s3:ObjectCreated:*'],
+                        'Filter' => new NotificationConfigurationFilter([
+                            'Key' => new S3KeyFilter([
+                                'FilterRules' => [
+                                    new FilterRule([
+                                        'Name' => 'prefix',
+                                        'Value' => 'change me',
+                                    ]),
+                                ],
+                            ]),
+                        ]),
+                    ]),
+                ],
+                'QueueConfigurations' => [
+                    new QueueConfiguration([
+                        'Id' => 'change me',
+                        'QueueArn' => 'change me',
+                        'Events' => ['s3:ObjectCreated:*'],
+                        'Filter' => new NotificationConfigurationFilter([
+                            'Key' => new S3KeyFilter([
+                                'FilterRules' => [new FilterRule([
+                                    'Name' => 'prefix',
+                                    'Value' => 'change me',
+                                ])],
+                            ]),
+                        ]),
+                    ]),
+                ],
+                'LambdaFunctionConfigurations' => [
+                    new LambdaFunctionConfiguration([
+                        'Id' => 'change me',
+                        'LambdaFunctionArn' => 'change me',
+                        'Events' => ['s3:ObjectCreated:*'],
+                        'Filter' => new NotificationConfigurationFilter([
+                            'Key' => new S3KeyFilter([
+                                'FilterRules' => [
+                                    new FilterRule([
+                                        'Name' => 'suffix',
+                                        'Value' => 'change me',
+                                    ]),
+                                ],
+                            ]),
+                        ]),
+                    ]),
+                ],
+            ]),
+            'ExpectedBucketOwner' => 'bucket-name',
+        ]);
+
+        // see example-1.json from SDK
+        $expected = '
+PUT /bucket-name?notification HTTP/1.0
+Content-Type: application/xml
+x-amz-expected-bucket-owner: bucket-name
+
+<?xml version="1.0" encoding="UTF-8"?>
+<NotificationConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
+    <TopicConfiguration>
+        <Id>change me</Id>
+        <Topic>change me</Topic>
+        <Event>s3:ObjectCreated:*</Event>
+        <Filter>
+            <S3Key><FilterRule><Name>prefix</Name><Value>change me</Value></FilterRule></S3Key>
+        </Filter>
+    </TopicConfiguration>
+    <QueueConfiguration>
+        <Id>change me</Id>
```

Comment:
I AWS documenation this Is loooks like an integer (https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketNotificationConfiguration.html)
Are you sure both string and int are allowed?

Reply:
In the examples they have both

```xml
<QueueConfiguration>
      <Id>1</Id>
</QueueConfiguration>
```

and

```xml
<CloudFunctionConfiguration>
    <Id>ObjectCreatedEvents</Id>
</CloudFunctionConfiguration>
```